business-networks-contracts/src/main/kotlin/net/corda/bn/schemas/MembershipStateSchemaV1.kt

```diff
@@ -27,10 +27,13 @@ object MembershipStateSchemaV1 : MappedSchema(schemaFamily = MembershipState::cl
     @Table(name = "membership_state")
     class PersistentMembershipState(
             @Column(name = "corda_identity")
-            val cordaIdentity: Party,
+            var cordaIdentity: Party? = null,
             @Column(name = "network_id")
-            val networkId: String,
+            var networkId: String = "",
             @Column(name = "status")
-            val status: MembershipStatus
-    ) : PersistentState()
+            var status: MembershipStatus = MembershipStatus.PENDING
+    ) : PersistentState() {
+        // Hibernate requires this no-argument constructor
+        constructor() : this(null, "", MembershipStatus.PENDING)
```

Comment:
Needed to fix these errors:
```
javax.persistence.PersistenceException: org.hibernate.InstantiationException: No default constructor for entity:  : net.corda.bn.schemas.MembershipStateSchemaV1$PersistentMembershipState
```

Reply:
If I look at other entities like DBTransaction in DBTransactionStorage it does not require a no arg ctor. Is the problem something else?

Reply:
Ok let me double check that

Reply:
You're right, the no-arg constructor wasn't needed.

Reply:
If you add nullable = false in the Column annotation does this help?
I don't think they should need defaults either which are then switched. Can we have val's with no initialisation.
Like in DBTransaction entity for example.

Reply:
Tried that, if I switch any of those parameters back to `val`, I get `Caused by: java.lang.NullPointerException: Parameter specified as non-null is null: method net.corda.core.node.services.VaultQueryException` - even if I add `nullable = false` to the Column annotation. Not sure what's happening here

Reply:
Hi @adelel1 just pushed a commit. Couldn't find a solution for the above issue - looks like those parameters need to be `var`s and need to be nullable too (could see the same pattern in other classes such as [VaultSchema](https://github.com/corda/corda/blob/2a2197508891fff7ede72cba7696083aacc17144/node/src/main/kotlin/net/corda/node/services/vault/VaultSchema.kt#L77)). If I don't make them nullable we're getting a VaultQueryException saying it's trying to assign a null value to a non-null type. Adding `nullable = true` to the column annotation also didn't help.